main.nf

```diff
@@ -1695,14 +1695,18 @@ process run_eklipse {
 	publishDir "${OUTDIR}/plots/mito", mode: 'copy', overwrite: 'true', pattern: '*.png'
 
 	input:
-		set group, id, file(bam), file(bai) from eklipse_bam
-
+		set group, id, file(bam), file(bai), sex, type from eklipse_bam.join(meta_eklipse, by: [0,1])
```

Comment:
Is the `[0, 1]` due to the `id` and `group` flipping? We should really sort that out soon ... Maybe we can replace all the group + id with a meta object. Then we could also bunch the `type` and `sex` into that one. An issue for another day though

Reply:
Maybe not so far away though. Seems like something to sort out before transitioning to DSL2

Reply:
It's not because of the flipping. It's just to say that it should match on both. The result is that the group and id channel does not become a wierd list. 
if I only match on group it will be:
group, id1, bam, bai, id1/2/3(random), sex, type.
if I only match on id it will be:
[group,group], id, bam, bai, sex, type

Reply:
I see!